src/core/mac/mac.cpp

```diff
@@ -2198,11 +2221,7 @@ otError Mac::HandleMacCommand(Frame &aFrame)
         mCounters.mRxBeaconRequest++;
         otLogInfoMac("Received Beacon Request");
 
-        if (mEnabled && (mBeaconsEnabled
-#if OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_WHEN_JOINABLE
-                         || IsBeaconJoinable()
-#endif // OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_WHEN_JOINABLE
-                             ))
+        if (mEnabled && CanSendBeacon())
```

Comment:
Rename to `ShouldSendBeacon()` and move the `mEnabled` check inside?

Reply:
Done. Thanks.